addons/l10n_es_edi_facturae/models/account_move_send.py

```diff
@@ -0,0 +1,458 @@
+            taxes_normal_computed = [tax for tax in taxes_computed["tax_lines_to_add"] if tax["tax_amount"] >= 0]
+
+            taxes_output = [_convert_computed_tax_to_template(self.env, tax) for tax in taxes_normal_computed]
+            totals['total_tax_outputs'] += sum((abs(tax["tax_amount"]) for tax in taxes_normal_computed))
+
+            tax_withheld_output = [_convert_computed_tax_to_template(self.env, tax) for tax in taxes_withheld_computed]
+            totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
+
+            invoice_line_values.update({
+                'ItemDescription': line.name,
+                'Quantity': line.quantity,
+                'UnitOfMeasure': line.product_uom_id.l10n_es_edi_facturae_uom_code,
+                'UnitPriceWithoutTax': total_price / line.quantity,
+                'TotalCost': total_price,
+                'DiscountsAndRebates': [{
+                    'DiscountReason': '',
+                    'DiscountRate': f'{line.discount:.2f}',
+                    'DiscountAmount': discount
+                }, ] if discount != 0. else [],
+                'Charges': [{
+                    'ChargeReason': '',
+                    'ChargeRate': f'{max(0, -line.discount):.2f}',
+                    'ChargeAmount': surcharge,
+                }, ] if surcharge != 0. else [],
+                'GrossAmount': gross_amount,
+                'TaxesOutputs': taxes_output,
+                'TaxesWithheld': tax_withheld_output,
+            })
+            items.append(invoice_line_values)
+            taxes += taxes_output
+            taxes_withheld += tax_withheld_output
+        return items, taxes, taxes_withheld, totals
+
+    @api.model
+    def _l10n_es_edi_facturae_export_facturae(self, invoice):
+        """
+        Produce the Facturae XML data for the invoice.
+
+        :param account.move invoice: The invoice to export.
+        :return: (data needed to render the full template, data needed to render the signature template)
+        """
+
+        company = invoice.company_id
+        partner = invoice.partner_id.parent_id or invoice.partner_id
+
+        if not company.vat:
+            raise UserError(_('The company needs a set tax identification number or VAT number'))
+
+        if not partner.vat:
+            raise UserError(_('The partner needs a set tax identification number or VAT number'))
+
+        def fmt_curr(value, curr_decimal_places):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.850...1).
+            return float_repr(value, curr_decimal_places)
+
+        eur_curr = self.env['res.currency'].search([('name', '=', 'EUR')])
+        inv_curr = invoice.currency_id or company.currency_id
+        legal_literals = invoice.narration.striptags() if invoice.narration else False
+        legal_literals = legal_literals.split(";") if legal_literals else False
```

Comment:
Wtf if that? There is no logic on the `narration` field. Maybe you should use `html2plaintext` instead?